src/parser.rs

```diff
@@ -1109,4 +1112,91 @@ CC,featureC
         );
     }
 
+    #[test]
+    fn test_cache_with_fully_cached_request() {
+        // Test for perfect vowpal-compatible hashing
+        let vw_map_string = r#"
+AA,featureA
+BB,featureB
+CC,featureC
+"#;
```

Comment:
did this go through cargo fmt? Formatting looks off here

Reply:
@bbenshalom I ran rustfmt on this file & this part isn't impact. As a follow up I can create a separate PR with only the run of `cargo fmt`

Reply:
[Linking](https://github.com/outbrain/fwumious_wabbit/pull/110) to PR